From 6255695bd9a64a083fd9027bc8e91802124c0b77 Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Tue, 13 Sep 2022 13:12:38 +0300
Subject: [PATCH] compulab: iot-gate-imx8plus: sb-iotgimx8plus: support
 revision 1.1

Add support for sb-iotgimx8plus baseboard rev.1.1:
 - EEPROM over I2C2 bus (base address 0x54)
 - SDIO (uSDHC2) inteface (4-bits mode)
 - User LED Green_1 GPIO pin fix
 - ECSPI2 bus minor fix

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 .../boot/dts/compulab/sb-iotgimx8plus.dtsi    | 20 +++++++++--
 .../dts/compulab/ucm-imx8m-plus-pinctrl.dtsi  | 36 +++++++++++++++++++
 .../boot/dts/compulab/ucm-imx8m-plus.dtsi     | 18 ++++++++--
 3 files changed, 69 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/compulab/sb-iotgimx8plus.dtsi b/arch/arm64/boot/dts/compulab/sb-iotgimx8plus.dtsi
index 288187032fa9..fe62c8166d1d 100644
--- a/arch/arm64/boot/dts/compulab/sb-iotgimx8plus.dtsi
+++ b/arch/arm64/boot/dts/compulab/sb-iotgimx8plus.dtsi
@@ -42,7 +42,7 @@
 		user-led-1-green {
 			label = "Green_1";
 			color = <LED_COLOR_ID_GREEN>;
-			gpios = <&gpio2 18 GPIO_ACTIVE_HIGH>;
+			gpios = <&gpio2 12 GPIO_ACTIVE_HIGH>;
 			default-state = "off";
 		};
 		user-led-1-red {
@@ -116,6 +116,14 @@
 	};
 };
 
+&i2c2 {
+	eeprom@54 {
+		compatible = "atmel,24c08";
+		reg = <0x54>;
+		pagesize = <16>;
+	};
+};
+
 &i2c3 {
 	status = "okay";
 
@@ -195,6 +203,10 @@
 
 &ecspi2 {
 	status = "okay";
+
+	spidev2_0: spidev2@0 {
+		status = "okay";
+	};
 };
 
 &i2c5 {
@@ -205,6 +217,10 @@
 	status = "okay";
 };
 
+&usdhc2 {
+	status = "okay";
+};
+
 /***** EXPANSION CONNECTOR - End *****/
 
 &iomuxc {
@@ -256,7 +272,7 @@
 
 	pinctrl_user_leds: userledsgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SD2_DATA3__GPIO2_IO18			0x19
+			MX8MP_IOMUXC_SD2_CD_B__GPIO2_IO12			0x19
 			MX8MP_IOMUXC_SPDIF_EXT_CLK__GPIO5_IO05			0x19
 			MX8MP_IOMUXC_SAI3_TXD__GPIO5_IO01			0x19
 			MX8MP_IOMUXC_SAI3_RXFS__GPIO4_IO28			0x19
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-pinctrl.dtsi b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-pinctrl.dtsi
index f782690243f4..9aa24bf02f14 100644
--- a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-pinctrl.dtsi
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-pinctrl.dtsi
@@ -191,6 +191,42 @@
 		>;
 	};
 
+	pinctrl_usdhc2: usdhc2grp {
+		fsl,pins = <
+			MX8MP_IOMUXC_SD2_CLK__USDHC2_CLK				0x190
+			MX8MP_IOMUXC_SD2_CMD__USDHC2_CMD				0x1d0
+			MX8MP_IOMUXC_SD2_DATA0__USDHC2_DATA0				0x1d0
+			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1				0x1d0
+			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2				0x1d0
+			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3				0x1d0
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT				0xc1
+		>;
+	};
+
+	pinctrl_usdhc2_100mhz: usdhc2grp-100mhz {
+		fsl,pins = <
+			MX8MP_IOMUXC_SD2_CLK__USDHC2_CLK				0x194
+			MX8MP_IOMUXC_SD2_CMD__USDHC2_CMD				0x1d4
+			MX8MP_IOMUXC_SD2_DATA0__USDHC2_DATA0				0x1d4
+			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1				0x1d4
+			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2				0x1d4
+			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3				0x1d4
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 			0xc1
+		>;
+	};
+
+	pinctrl_usdhc2_200mhz: usdhc2grp-200mhz {
+		fsl,pins = <
+			MX8MP_IOMUXC_SD2_CLK__USDHC2_CLK				0x196
+			MX8MP_IOMUXC_SD2_CMD__USDHC2_CMD				0x1d6
+			MX8MP_IOMUXC_SD2_DATA0__USDHC2_DATA0				0x1d6
+			MX8MP_IOMUXC_SD2_DATA1__USDHC2_DATA1				0x1d6
+			MX8MP_IOMUXC_SD2_DATA2__USDHC2_DATA2				0x1d6
+			MX8MP_IOMUXC_SD2_DATA3__USDHC2_DATA3				0x1d6
+			MX8MP_IOMUXC_GPIO1_IO04__USDHC2_VSELECT 			0xc1
+		>;
+	};
+
 	pinctrl_usdhc3: usdhc3grp {
 		fsl,pins = <
 			MX8MP_IOMUXC_NAND_WE_B__USDHC3_CLK				0x190
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus.dtsi b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus.dtsi
index f1a56d6ee072..f2db6777a48f 100644
--- a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus.dtsi
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus.dtsi
@@ -61,14 +61,15 @@
 	#size-cells = <0>;
 	fsl,spi-num-chipselects = <1>;
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_ecspi2 &pinctrl_ecspi2_cs>;
+	pinctrl-0 = <&pinctrl_ecspi2>, <&pinctrl_ecspi2_cs>;
 	cs-gpios = <&gpio5 13 GPIO_ACTIVE_LOW>;
-	status = "okay";
+	status = "disabled";
 
-	spidev1: spi@0 {
+	spidev2_0: spidev2@0 {
 		reg = <0>;
 		compatible = "rohm,dh2228fv";
 		spi-max-frequency = <500000>;
+		status = "disabled";
 	};
 };
 
@@ -405,6 +406,17 @@
 	status = "disabled";
 };
 
+&usdhc2 {
+	assigned-clocks = <&clk IMX8MP_CLK_USDHC2>;
+	assigned-clock-rates = <400000000>;
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2>;
+	pinctrl-1 = <&pinctrl_usdhc2_100mhz>;
+	pinctrl-2 = <&pinctrl_usdhc2_200mhz>;
+	bus-width = <4>;
+	status = "disabled";
+};
+
 &usdhc3 {
 	assigned-clocks = <&clk IMX8MP_CLK_USDHC3>;
 	assigned-clock-rates = <400000000>;
-- 
2.20.1

